aim/api/infra.py

```diff
@@ -76,14 +76,17 @@ class HostDomainMapping(resource.ResourceBase):
     """host to VMM and phys-dom mapping"""
 
     identity_attributes = t.identity(
-        ('host_name', t.string(128)))
-    other_attributes = t.other(
-        ('vmm_domain_name', t.string(64)),
-        ('physical_domain_name', t.string(64)))
+        ('host_name', t.string(128)),
+        ('domain_name', t.string(64)),
+        ('domain_type', t.enum('PhysDom',
+                               'OpenStack',
+                               'Kubernetes',
+                               'VMware')))
+    other_attributes = t.other()
 
     def __init__(self, **kwargs):
-        super(HostDomainMapping, self).__init__({'vmm_domain_name': '',
-                                                 'physical_domain_name': ''},
+        super(HostDomainMapping, self).__init__({'domain_name': '*',
```

Comment:
The '*' should be in the host_name field instead though. But even if you do that, I think the "aimctl manager" command is not going to work well as its expecting 3 parameters so it doesn't know which one is which one say if you just provide 2 parameters there.....Chatted with Amit a bit about this. He said if you really want to provide a default value for the primary key parameters then you also have to modify the CLI parser code which is not trivial. So here I will suggest not to provide any default values for all these parameters as they are all primary key.

Reply:
Thanks for finding that out! Pushed an update that reverts/removes the defaults, and fixes UT issue.